### PR TITLE
Zookeeper TX Log Cleanup

### PR DESCRIPTION
**Pull request by dealako**
* Added task to create a Zookeeper cleanup cron job that runs daily at
midnight
* Ignore VSC project files

JIRA: RE-2158

Issue: [RE-2158](https://rpc-openstack.atlassian.net/browse/RE-2158)

### CODE REVIEW

File: nodepool/setup_nodepool.yml [hunk 1]
```diff
@@ -119,6 +119,15 @@
...
         name: openstack.logrotate
 
+    - name: Zookeeper TX Log Cleanup Cron
+      cron:
+        name: "zookeeper cleanup"
+        minute: 0
+        hour: 0
+        user: root
+        job: "/usr/share/zookeeper/bin/zkCleanup.sh -n 3 >> /var/log/zookeeper/zookeeper_cron_cleanup.log"
+        cron_file: "zokeeper_cleanup"
```
Comment (odyssey4me) [inline nodepool/setup_nodepool.yml]: spelling error here

File: nodepool/setup_nodepool.yml [hunk 2]
```diff
@@ -119,6 +119,15 @@
       include_role:
         name: openstack.logrotate
 
+    - name: Zookeeper TX Log Cleanup Cron
+      cron:
+        name: "zookeeper cleanup"
+        minute: 0
+        hour: 0
+        user: root
+        job: "/usr/share/zookeeper/bin/zkCleanup.sh -n 3 >> /var/log/zookeeper/zookeeper_cron_cleanup.log"
```
Comment (odyssey4me) [inline nodepool/setup_nodepool.yml]: This log file should be added to the log rotation config.

Comment (dealako) [reply nodepool/setup_nodepool.yml]: Good catch! Added to log rotation as weekly.

File: nodepool/setup_nodepool.yml [hunk 3]
```diff
@@ -119,6 +127,15 @@
...
```
Comment (odyssey4me) [inline nodepool/setup_nodepool.yml]: Perhaps this should make sure that both stdout and stderr go to the log file?

**Comment by dealako** [comment]
recheck_unit_tests

**Pull request merged**